### PR TITLE
feat: switch controller-runtime to zap.Logger

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/talos/issues/3489
Enable logging using default development config with some fine tuning.
Additionally, now `info` and below logs go to kmsg.

Console logs:

```
10.5.0.2: [talos] task labelNodeAsMaster (1/1): starting
10.5.0.2: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.ManifestApplyController", "error": "error creating mapping for object /v1/Secret/bootstrap-token-uab6tj: an error on the server (\"\") has prevented the request from succeeding"}
10.5.0.2: [talos] retrying error: an error on the server ("") has prevented the request from succeeding (get nodes talos-default-master-1)
10.5.0.2: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.KubeletStaticPodController", "error": "error refreshing pod status: error fetching pod status: an error on the server (\"Authorization error (user=apiserver-kubelet-client, verb=get, resource=nodes, subresource=proxy)\") has prevented the request from succeeding"}
10.5.0.2: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.ManifestApplyController", "error": "error creating mapping for object /v1/Secret/bootstrap-token-uab6tj: an error on the server (\"\") has prevented the request from succeeding"}
10.5.0.2: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.ManifestApplyController", "error": "error creating mapping for object /v1/Secret/bootstrap-token-uab6tj: an error on the server (\"\") has prevented the request from succeeding"}
10.5.0.2: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.KubeletStaticPodController", "error": "error refreshing pod status: error fetching pod status: an error on the server (\"Authorization error (user=apiserver-kubelet-client, verb=get, resource=nodes, subresource=proxy)\") has prevented the request from succeeding"}
10.5.0.2: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.ManifestApplyController", "error": "error creating mapping for object /v1/Secret/bootstrap-token-uab6tj: an error on the server (\"\") has prevented the request from succeeding"}
```

Service logs:
```
cceeding"}
10.5.0.2: 2021-04-19T15:48:01.328Z DEBUG restarting controller in 843.619967ms {"component": "controller-runtime", "controller": "k8s.KubeletStaticPodController"}
10.5.0.2: 2021-04-19T15:48:02.173Z DEBUG controller starting {"component": "controller-runtime", "controller": "k8s.KubeletStaticPodController"}
10.5.0.2: 2021-04-19T15:48:07.909Z ERROR controller failed {"component": "controller-runtime", "controller": "k8s.ManifestApplyController", "error": "error creating mapping for object /v1/Secret/bootstrap-token-uab6tj: an error on the server (\"\") has prevented the request from succeeding"}
10.5.0.2: 2021-04-19T15:48:07.922Z DEBUG restarting controller in 1.221625364s {"component": "controller-runtime", "controller": "k8s.ManifestApplyController"}
10.5.0.2: 2021-04-19T15:48:09.145Z DEBUG controller starting {"component": "controller-runtime", "controller": "k8s.ManifestApplyController"}
10.5.0.2: 2021-04-19T15:48:09.159Z DEBUG waiting for mutex {"component": "controller-runtime", "controller": "k8s.ManifestApplyController"}
10.5.0.2: 2021-04-19T15:48:09.167Z DEBUG mutex acquired {"component": "controller-runtime", "controller": "k8s.ManifestApplyController"}
10.5.0.2: 2021-04-19T15:48:17.231Z WARN Warning: policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+ {"component": "controller-runtime", "controller": "k8s.ManifestApplyController"}
10.5.0.2: 2021-04-19T15:49:43.828Z INFO adjusting time by 2.299636ms via 80.249.145.122, state TIME_OK, status STA_NANO | STA_PLL {"component": "controller-runtime", "controller": "time.SyncController"}
```

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: internal/app/machined/pkg/runtime/logging/zap.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (smira) [inline internal/app/machined/pkg/runtime/logging/zap.go:1]: I think we can move this several levels up to `pkg/logging`, as these functions might be useful for other components as well? 

they seem to be pretty generic and really not specific to machined/runtime

**Review by smira — DISMISSED** [review]
:+1:

**Review by Ulexus — APPROVED** [review]

**Pull request merged**